app/dao/fact_notification_status_dao.py

```diff
@@ -260,7 +260,7 @@ def fetch_notification_status_for_service_for_today_and_7_previous_days(service_
             *([func.count().label("count")]),
         )
         .filter(
-            Notification.created_at >= start_date,
+            Notification.created_at >= get_midnight(datetime.utcnow()),
```

Comment:
This line was the core of the issue that inflated the notification counts. We weren't pulling just today's notifications in this query, but all notifications from starting from `today - days of  retention` to `today`.